jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/BaseKDisjointShortestPathsAlgorithm.java

```diff
@@ -0,0 +1,255 @@
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Iterate over all paths to remove overlapping edges (i.e. those edges contained in more than 
+     * one path).
+     * Two edges are considered as overlapping in case both edges connect the same vertex pair, 
+     * disregarding direction.
+     * At the end of this method, each path contains unique edges but not necessarily connecting the
+     * start to end vertex.
+     * 
+     */
+    private void findOverlappingEdges()
+    {
+        Map<UnorderedPair<V, V>, Integer> edgeOccurrenceCount = new HashMap<>();
+        for (List<E> path : pathList) {
+            for (E e : path) {                
+                V v = this.getEdgeSource(e);
+                V u = this.getEdgeTarget(e);                
+                UnorderedPair<V, V> edgePair = new UnorderedPair<>(v, u);
+                
+                if (edgeOccurrenceCount.containsKey(edgePair)) {
+                    edgeOccurrenceCount.put(edgePair, 2);
+                } else {
+                    edgeOccurrenceCount.put(edgePair, 1);
+                }
+            }
+        }
+
+        this.overlappingEdges = pathList.stream().flatMap(List::stream).filter(
+            e -> edgeOccurrenceCount.get(new UnorderedPair<>(
+                this.getEdgeSource(e), 
+                this.getEdgeTarget(e))) > 1)
+            .collect(Collectors.toSet());
+    }
+
+    private GraphPath<V, E> createGraphPath(List<E> edgeList, V startVertex, V endVertex)
+    {
+        double weight = 0;
+        for (E edge : edgeList) {
+            weight += originalGraph.getEdgeWeight(edge);
+        }
+        return new GraphWalk<>(originalGraph, startVertex, endVertex, edgeList, weight);
+    }
+    
+    private V getEdgeSource(E e) {
+        return this.workingGraph.containsEdge(e) ? this.workingGraph.getEdgeSource(e) : this.originalGraph.getEdgeSource(e);
+    }
+    
+    private V getEdgeTarget(E e) {
+        return this.workingGraph.containsEdge(e) ? this.workingGraph.getEdgeTarget(e) : this.originalGraph.getEdgeTarget(e);
+    }
+        
+    /**
+     * Calculates the shortest paths for the current iteration.
+     * Path is not final and to be used in a "post-production" phase
+     * (see resolvePaths method).
+     * 
+     * @param startVertex the start vertex
+     * @param endVertex the end vertex
+     * 
+     * @return the shortest path between start and end vertices.
+     */
+    protected abstract GraphPath<V, E> calculateShortestPath(V startVertex, V endVertex);
+    
+    /**
+     * Prepares the working graph for next iteration. To be called from the second iteration
+     *  and on, so implementation may assume a preceding {@link #calculateShortestPath} call.
+     *  
+     * @param previousPath the path found at the previous iteration.
+     */
+    protected abstract void prepare(List<E> previousPath);

```

Comment:
I still think a name like transformPreviousPath is much clearer than prepare.

Reply:
👍🏼